### PR TITLE
feat : allow dropdown menu on toolbar to group secondary buttons

### PR DESCRIPTION
**Pull request by firm1**
This PR allow to groups somes buttons in dropdown menu.

Result give something like screenshot

Capture d’écran de 2020-01-22 22-57-00

### CODE REVIEW

File: src/js/easymde.js [hunk 1]
```diff
@@ -110,15 +110,30 @@ function fixShortcut(name) {
     return name;
 }
 
+/**
+ * Create dropdown block
+ */
+function createToolbarDropdown(options, enableTooltips, shortcuts, parent) {
+    var el = createToolbarButton(options, enableTooltips, shortcuts, 'div');
```
Comment (Situphen) [inline src/js/easymde.js]: I think you forgot to give `parent` to the function `createToolbarButton(options, enableTooltips, shortcuts, markup, parent)`

File: src/js/easymde.js [hunk 2]
```diff
@@ -2252,24 +2281,14 @@ EasyMDE.prototype.createToolbar = function (items) {
...
-                    el.onclick = function (e) {
-                        e.preventDefault();
-                        item.action(self);
-                    };
-                } else if (typeof item.action === 'string') {
-                    el.onclick = function (e) {
-                        e.preventDefault();
-                        window.open(item.action, '_blank');
-                    };
+                if (item.children) {
```
Comment (Situphen) [inline src/js/easymde.js]: I think `else if` would be easier to read than `else` followed by `if`.

File: src/css/easymde.css [hunk 3]
```diff
@@ -316,3 +313,20 @@
...
+.easymde-dropdown {
+    position: relative;
+}
+
+.easymde-dropdown-content {
+    display: none;
+    position: absolute;
+    background-color: #f9f9f9;
+    box-shadow: 0px 8px 16px 0px rgba(0,0,0,0.2);
+    padding: 8px 8px;
```
Comment (Situphen) [inline src/css/easymde.css]: Why not simply `padding: 8px;`?

**Comment by firm1** [comment]
Thank you @Situphen for your review, I updated the code with your suggests.

@lonaru , you can decide now.

**Comment by A-312** [comment]
fix: #129 *

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Group buttons in drop-down lists by adding a sub-option `children` for the items in the toolbar.
```
Comment (Situphen) [inline CHANGELOG.md]:
```suggestion
- Group buttons in drop-down lists by adding a sub-option `children` for the items in the toolbar (Thanks to [@firm1], [#141]).
```

**Comment by firm1** [comment]
Very good remarks, thanks.  I'll take care of your suggestions.

**Comment by A-312** [comment]
2) We can use `▼` on `::before` or `::after`

**Comment by firm1** [comment]
Hi @Ionaru ,
I've considered all your comments, can you check that it corresponds to what is expected ?

Capture d’écran de 2020-01-28 18-58-12


Cheers,

**Comment by firm1** [comment]
Else, if you prefer, i have this style too.

Capture d’écran de 2020-01-28 21-33-12


Capture d’écran de 2020-01-28 21-33-06

**Comment by Ionaru** [comment]
A small thing I've noticed is that the `noDisable` flag doesn't get applied properly, the button is no longer clickable when in preview-mode, but the cursor still turns into a pointer when hovering over the dropdown.
image

**Comment by firm1** [comment]
I @Ionaru i can't reproduce your last problem. 

Can you give me your html example file and your browser name and version please ?

**Comment by Ionaru** [comment]
```js
    new EasyMDE({
        autoDownloadFontAwesome: false,
        toolbar: [{
            name: 'bold',
            className: 'fa fa-bold',
            title: 'Bold',
            noDisable: true,
            children: [
                {
                    name: "italic",
                    action: EasyMDE.toggleItalic,
                    className: "fa fa-italic",
                    title: "Italic",
                }
            ]
        }]
    });
```
Then enable preview mode and hover over the `italic` button.

**Comment by A-312** [comment]
@Ionaru In the master branch, I get the same behavior of this PR 
image

**Comment by A-312** [comment]
ping @Ionaru can you explain more that we should get ? because on master branch I got the same behavior

**Comment by Ionaru** [comment]
Hmm alright if it's the same on master then it's fine I suppose

**Comment by A-312** [comment]
> Hmm alright if it's the same on master then it's fine I suppose

If you want that we fix this issue, we need to understand the behavior of this property. But we are waiting this PR, maybe we can open a ticket ?

**Comment by Ionaru** [comment]
@A-312 @firm1 I've made a small change to the dropdown code to allow the dropdown to use the built-in button shortcuts.

```js
new EasyMDE({
    toolbar: [{
        name: 'Text tools',
        className: 'fas fa-edit',
        title: 'Common text edit tools',
        noDisable: true,
        children: ['bold', 'italic'],
    }],
});
```

Please check if the code is also still valid for your use-cases.

**Comment by A-312** [comment]
firm1 has less time to watch it, I will watch this evening.

**Comment by A-312** [comment]
I have forgotten to review this PR, I did, this seem OK for me.

**Pull request merged**